### PR TITLE
Introduce `core.gardener.cloud/v1alpha1.Project` resource

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
This PR creates the new `core.gardener.cloud/v1alpha1.Project` resource together with forwards and backwards compatibility compared to the current `garden.sapcloud.io/v1beta1.Project` resources, based on [GEP-4](#1284).

**Special notes for your reviewer**:
* Part of #1376
* The gardener-controller-manager is still working on `garden.sapcloud.io/v1beta1.Project`, I suggest to not change this until we have introduced all resources in the new `core` API group and then change the GCM to use this API group in one go.

**Release note**:

```noteworthy user
Gardener does now feature a new `Project` resource which is part of the `core.gardener.cloud/v1alpha1` API group. It is fully forwards and backwards compatible to the old `Project` resource in the `garden.sapcloud.io/v1beta1` API group. It will be the new default, and the old `garden.sapcloud.io/v1beta1.Project` resource are deprecated now. It will be removed in a future version. Consider switching to using the new `core.gardener.cloud/v1alpha1.Project` resource. The `example` directory contains proper example manifests.
```

### CODE REVIEW

File: docs/proposals/04-new-core-gardener-cloud-apis.md
Comment (vpnachev) [inline docs/proposals/04-new-core-gardener-cloud-apis.md]:
```suggestion
apiVersion: core.gardener.cloud/v1alpha1
```

File: docs/proposals/04-new-core-gardener-cloud-apis.md
Comment (vpnachev) [inline docs/proposals/04-new-core-gardener-cloud-apis.md]: #### question

do we need this annotation for the new API?
If yes, I am confused because the project name is `example` and `spec.namespace` is `garden-example`, but the annotation value is `garden-core`.

File: docs/proposals/04-new-core-gardener-cloud-apis.md
Comment (vpnachev) [inline docs/proposals/04-new-core-gardener-cloud-apis.md:289]: #### comment

I would keep the role name `member`, not `admin`.
We might decide to narrow the privileges of the members(disallow creation/reading of secret/secretBindings for example and let only admins maintain cloud credentials).

But honestly, I have no strong opinion about this because we can introduce new roles latter that will have different access levels.

File: pkg/apis/core/v1alpha1/types_project.go
Comment (vpnachev) [inline pkg/apis/core/v1alpha1/types_project.go]: > or any other identifier of a user

This can be a `Group` or `ServiceAccount` name, not only user.

**Review by vpnachev — CHANGES_REQUESTED** [review]
just some minor nits and comments.

**Comment by vpnachev** [comment]
@gardener/dashboard-maintainers should be also interested in this change.

File: docs/proposals/04-new-core-gardener-cloud-apis.md
Comment (rfranzke) [reply docs/proposals/04-new-core-gardener-cloud-apis.md]: No, you're right, thanks, I remove it.

File: docs/proposals/04-new-core-gardener-cloud-apis.md
Comment (rfranzke) [reply docs/proposals/04-new-core-gardener-cloud-apis.md:289]: Hm, valid suggestion/argument, but `member` role inside the `members[]` list is little confusing and not very explicit. I'd keep it as `admin`, because that's what's implemented today: Admins have full access (wrt to what is needed to interact with Gardener API) in their project.

Comment (vpnachev) [reply docs/proposals/04-new-core-gardener-cloud-apis.md:289]: Yes, I agree.
We can say that previously we wrongly gave the name 'members' to users that has 'admin' privileges, so now it will be fixed - they will have the role `admin` and keep the same privileges.

File: pkg/apis/core/v1alpha1/types_project.go
Comment (vpnachev) [inline pkg/apis/core/v1alpha1/types_project.go]:
```suggestion
	// Subject is representing a user name, an email address, or any other identifier of a user, group, or service account that has a certain role.
```

File: pkg/apis/garden/types.go
Comment (vpnachev) [inline pkg/apis/garden/types.go]:
```suggestion
	// group, or service account that has a certain role.
```

File: pkg/apis/garden/types.go
Comment (vpnachev) [inline pkg/apis/garden/types.go]:
```suggestion
	// Members is a list of subjects representing a user name, an email address, or any other identifier of a user,
```

File: pkg/apis/garden/types.go
Comment (vpnachev) [inline pkg/apis/garden/types.go]:
```suggestion
	// Subject is representing a user name, an email address, or any other identifier of a user, group, or service account that has a certain role.
```

File: pkg/openapi/openapi_generated.go
Comment (vpnachev) [inline pkg/openapi/openapi_generated.go]: re-generate code to update the open API documentation

**Review by vpnachev — CHANGES_REQUESTED** [review]
Thanks!
Just apply the documentation improvements on couple more places and regenerated the code.

**Review by vpnachev — DISMISSED** [review]
Thanks!

**Comment by rfranzke** [comment]
Thanks for the review!
FYI: I just added https://github.com/gardener/gardener/pull/1382/files#diff-d4f36957532fc8dd5890f3f339d95601R116-R123.

Will merge once CI passed.

**Review by vpnachev — APPROVED** [review]

**Pull request merged**
